Packs/EDL/Integrations/EDL/EDL.py

```diff
@@ -785,35 +789,54 @@ def get_outbound_mimetype(request_args: RequestArguments) -> str:
 
 
 @debug_function
-def get_edl_on_demand():
+def get_edl_on_demand() -> tuple[str, int]:
     """
     Use the local file system to store the on-demand result, using a lock to
     limit access to the file from multiple threads.
+
+    Returns:
+        tuple[str, int]: A tuple of formatted indicators to display in EDL's response (str),
+            and the number of original indicators received from the server before formatting (int).
     """
+    global EDL_ON_DEMAND_CACHE_ORIGINAL_SIZE
     ctx = get_integration_context()
+
     if EDL_ON_DEMAND_KEY in ctx:
         ctx.pop(EDL_ON_DEMAND_KEY, None)
         request_args = RequestArguments.from_context_json(ctx)
-        edl = create_new_edl(request_args)
+        edl_data, original_indicators_count = create_new_edl(request_args)
```

Comment:
let's remove `original_indicators_count` to simplify the code flow, and use just `EDL_ON_DEMAND_CACHE_ORIGINAL_SIZE` - even if `file.write(edl_data)` fails, it'll have to recreate the EDL, so it's the same effect.